cluster_configs/barnacle/barnacle_status.py

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python
+
+import subprocess
+import sys
+import time
+import re
+from io import StringIO
+
+def parse_qstat(lines):
+    q_dict = {}
+
+    p = re.compile('^ +(.+?) = (.+)')
```

Comment:
I found that regular expressions in python are terrible slow. Thus, if somehow possible you should avoid them if performance is important. I don't know how many lines you have to parse here, if it is < 1000 it should be fine as is

Reply:
It's parsing [these status reports](https://github.com/tanaes/oecophylla/blob/1c43b6ae2084211a14a4faa4f10989d22afe0937/cluster_configs/barnacle/tests/test_data/success.txt) from qstat, so like 60-100 lines. However, it is doing it a few times per second per simultaneous running job. Do you think that's fine?